ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -646,7 +649,12 @@ withTestMempool setup@TestSetup { testLedgerState, testInitialTxs, testMempoolCa
     cfg = ledgerConfigView singleNodeTestConfig
 
     setUpAndRun :: forall m. IOLike m => m Property
-    setUpAndRun = do
+    setUpAndRun = withRegistry $ \registry -> do
+
+      -- Set up the BlockchainTime
+      TestBlockchainTime
+        { testBlockchainTime
+        } <- newTestBlockchainTime registry (NumSlots 1) (secondsToDiffTime 0)
```

Comment:
In PR #1176, this will be replaced with a new type of `BlockchainTime`, `ManualBlockchainTime`, that can be manually ticked.

Reply:
BTW, do you really need a separate type for it? I have written the following:
```haskell
settableBlockchainTime :: IOLike m => StrictTVar m SlotNo -> BlockchainTime m
settableBlockchainTime varCurSlot = BlockchainTime {
      getCurrentSlot = readTVar varCurSlot
    , onSlotChange_  = error "unimplemented onSlotChange_"
    }
```
If you need `onSlotChange_`, you'll need to do some more work, though.

Reply:
`onSlotChange_` is also easy, just use `onEachChange`.

Reply:
Indeed, the real block chain time is not much more than that; it just has an additional thread that updates the TVar when the wallclock changes. Could use this in the implementation of `realBlockchainTime`.